src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java

```diff
@@ -43,82 +44,81 @@ public HyperwalletApiClient(final String username, final String password, final
-            return processResponse(response, type);
+            Request request = getService(url, true);
+            Response response = request.getResource();
+            return processResponse(request, response, type);
         } catch (IOException | JOSEException | ParseException e) {
             throw new HyperwalletException(e);
         }
     }
 
     public <T> T put(final String url, final Object bodyObject, final Class<T> type) {
-        Response response = null;
         try {
-            String body = convert(bodyObject);
-            response = getService(url, false).setBody(encrypt(body)).putResource();
-            return processResponse(response, type);
+            String body = encrypt(convert(bodyObject));
+            Request request = getService(url, false).setBody(body);
+            Response response = request.putResource();
+            return processResponse(request, response, type);
         } catch (IOException | JOSEException | ParseException e) {
             throw new HyperwalletException(e);
         }
     }
 
     public <T> T post(final String url, final Object bodyObject, final Class<T> type) {
-        Response response = null;
         try {
-            Request request = getService(url, false);
             String body = bodyObject != null ? encrypt(convert(bodyObject)) : "";
-            request.setBody(body);
-            response = request.postResource();
-            return processResponse(response, type);
+            Request request = getService(url, false).setBody(body);
+            Response response = request.postResource();
+            return processResponse(request, response, type);
         } catch (IOException | JOSEException | ParseException e) {
             throw new HyperwalletException(e);
         }
     }
 
     public <T> T post(final String url, final Object bodyObject, final Class<T> type, HashMap<String,String> header) {
-        Response response = null;
         try {
-            String body = convert(bodyObject);
-            Request request = getService(url, false).setBody(encrypt(body));
+            String body = encrypt(convert(bodyObject));
+            Request request = getService(url, false).setBody(body);
             if (header != null) {
                 for (String key : header.keySet()) {
                     request = request.addHeader(key, header.get(key));
                 }
             }
 
-            response = request.postResource();
-            return processResponse(response, type);
+            Response response = request.postResource();
+            return processResponse(request, response, type);
         } catch (IOException | JOSEException | ParseException e) {
             throw new HyperwalletException(e);
         }
     }
 
-    protected <T> T processResponse(final Response response, final Class<T> type)
+    protected <T> T processResponse(final Request request, final Response response, final Class<T> type)
```

Comment:
This method is like very complicated why do we process request object here? when the method itself is processResponse ?